src/mpid/ch4/netmod/ofi/ofi_pre.h

```diff
@@ -171,7 +171,9 @@ typedef struct {
 
 typedef struct {
     fi_addr_t dest;
-#if defined MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS || defined MPIDI_OFI_ENABLE_RUNTIME_CHECKS
+#if MPIDI_OFI_ENABLE_RUNTIME_CHECKS
+    unsigned ep_idx:MPIDI_OFI_MAX_ENDPOINTS_BITS_SCALABLE;
+#elif MPIDI_OFI_ENABLE_SCALABLE_ENDPOINTS
```

Comment:
The if(directive) stuff is fine but then I think you have to take all the #if stuff out of the data structures